SPECS/mock-core-configs/mock-core-configs.spec

```
@@ -62,15 +84,64 @@ fi
 # reference valid mock.rpm's docdir with example site-defaults.cfg
 mock_docs=%{_pkgdocdir}
 mock_docs=${mock_docs//mock-core-configs/mock}
-mock_docs=${mock_docs//-%{version}/-*}
+mock_docs=${mock_docs//-%version/-*}
 sed -i "s~@MOCK_DOCS@~$mock_docs~" %{buildroot}%{_sysconfdir}/mock/site-defaults.cfg
 
 %post
-if [ -s %{_sysconfdir}/os-release ]; then
-    ver=$(source %{_sysconfdir}/os-release && echo $VERSION_ID | cut -d. -f1 | grep -o '[0-9]\+')
+if [ -s /etc/os-release ]; then
+    # fedora and rhel7+
+    if grep -Fiq Rawhide /etc/os-release; then
+        ver=rawhide
+    # mageia
+    elif [ -s /etc/mageia-release ]; then
+        if grep -Fiq Cauldron /etc/mageia-release; then
+           ver=cauldron
+        fi
+    else
+        ver=$(source /etc/os-release && echo $VERSION_ID | cut -d. -f1 | grep -o '[0-9]\+')
+    fi
+else
+    # something obsure, use buildtime version
+    ver=%{?rhel}%{?fedora}%{?mageia}
+fi
+if [ -s /etc/mageia-release ]; then
+    mock_arch=$(sed -n '/^$/!{$ s/.* \(\w*\)$/\1/p}' /etc/mageia-release)
+else
+    mock_arch=$(%{python3} -c "import dnf.rpm; import hawkey; print(dnf.rpm.basearch(hawkey.detect_arch()))")
 fi
-mock_arch=$(python -c "import rpmUtils.arch; baseArch = rpmUtils.arch.getBaseArch(); print baseArch")
-cfg=%{?fedora:fedora}%{?rhel:epel}%{?mageia:mageia}-$ver-${mock_arch}.cfg
+
+cfg=unknown-distro
+%if 0%{?fedora}
+cfg=fedora-$ver-$mock_arch.cfg
+%endif
+%if 0%{?azl}
+cfg=azurelinux-3.0-$mock_arch.cfg
```

We shouldn't hard-code a version in the spec:
```suggestion
cfg=azurelinux-%{azl}.0-$mock_arch.cfg
```

---

Or, since the `ver` variable seems to be derived for this exact purpose:
```suggestion
cfg=azurelinux-%ver-$mock_arch.cfg
```

---

FYI, `$ver` doesn't work as-is because of the additional `.0`, but I'd agree `$ver.0` is still cleaner. I'll go with that unless you feel strongly otherwise.

---

No, this is what I was thinking about, just didn't notice `$ver` is just the major version number.